backend/src/main/kotlin/org/loculus/backend/service/submission/CurrentProcessingPipelineTable.kt

```diff
@@ -1,11 +1,42 @@
 package org.loculus.backend.service.submission
 
+import kotlinx.datetime.LocalDateTime
 import org.jetbrains.exposed.sql.Table
+import org.jetbrains.exposed.sql.andWhere
+import org.jetbrains.exposed.sql.batchInsert
 import org.jetbrains.exposed.sql.kotlin.datetime.datetime
+import org.jetbrains.exposed.sql.selectAll
+import org.jetbrains.exposed.sql.update
 
 const val CURRENT_PROCESSING_PIPELINE_TABLE_NAME = "current_processing_pipeline"
 
 object CurrentProcessingPipelineTable : Table(CURRENT_PROCESSING_PIPELINE_TABLE_NAME) {
+    val organismColumn = varchar("organism", 255)
     val versionColumn = long("version")
     val startedUsingAtColumn = datetime("started_using_at")
+
+    fun setV1ForOrganismsIfNotExist(organisms: Collection<String>, now: LocalDateTime) =
+        CurrentProcessingPipelineTable.batchInsert(organisms, ignore = true) { organism ->
+            this[organismColumn] = organism
+            this[versionColumn] = 1
+            this[startedUsingAtColumn] = now
+        }
+
+    fun pipelineNeedsUpdate(foundVersion: Long, organism: String) = CurrentProcessingPipelineTable
+        .selectAll()
+        .where { versionColumn neq foundVersion }
+        .andWhere { organismColumn eq organism }
+        .limit(1)
+        .empty()
+        .not()
```

Comment:
I think that we need to check whether the found version is larger than all existing versions?

Reply:
Yes! Fixed: https://github.com/loculus-project/loculus/pull/3534/commits/fdc37b286142a7b22df39f78968dbf44e4af780c

(It worked before due to constraints elsewhere in the codebase, but this is more robust and correct 👍 )